server/src/main/java/org/opensearch/search/internal/SearchContext.java

```diff
@@ -366,6 +366,13 @@ public final void assignRescoreDocIds(RescoreDocIds rescoreDocIds) {
      */
     public abstract Profilers getProfilers();
 
+    /**
+     * Returns concurrent segment search status for the search context
+     */
+    public boolean isConcurrentSegmentSearchEnabled() {
+        return false;
+    }
```

Comment:
I left this default implementation here because there are a few other `SearchContext` types and didn't want to implement it in each class.